src/wp-admin/includes/export.php

```diff
@@ -144,6 +144,40 @@ function export_wp( $args = array() ) {
 	// Grab a snapshot of post IDs, just in case it changes during the export.
 	$post_ids = $wpdb->get_col( "SELECT ID FROM {$wpdb->posts} $join WHERE $where" );
 
+	// Get IDs for the attachments of each post, unless all content is already being exported.
+	if ( 'all' !== $args['content'] ) {
```

Comment:
Those extra queries will be executed when exporting attachments, where it is not useful.

We could update the condition to skip this part if the `$args['content']` is set to `all` or `attachment`. I think it would be safe to hard-code those two values, but we could also consider adding a filter to allow developers to include custom post types to the list.

Reply:
>We could update the condition to skip this part if the $args['content'] is set to all or attachment.

+1

Reply:
>could also consider adding a filter to allow developers to include custom post types to the list.

Yes, a filter could be added for customization. That said, adding a filter would need to wait until 6.6, as it's too late in the 6.5 cycle given adding a filter is considered an `enhancement` and 6.5 is now in beta.

Reply:
I updated the condition to skip if the export is `all` or `attachment`. I did it in a way that will make it easy to add a filter in the future. We would just filter the `array( 'all', 'attachment' )` array.